falcon/hooks.py

```diff
@@ -82,14 +95,14 @@ def let(responder=responder):
 
         else:
             responder = responder_or_resource
-            do_before_one = _wrap_with_before(responder, action, args, kwargs)
+            do_before_one = _wrap_with_before(responder, action, args, kwargs, is_async)
 
             return do_before_one
 
     return _before
 
 
-def after(action, *args, **kwargs):
+def after(action, *args, is_async=None, **kwargs):
```

Comment:
I will say, however, that this can break at least one calling pattern when the default is not `False` (not a problem in this case). If the caller pre-composes a kwargs dict, for example, and wants to just pass through an option `bar` from upstream that does default to `None` for whatever reason, they can not always just set {'foo': bar}, but will have to add additional logic to avoid including the kwarg in the dict in order to be sure they trigger the desired default value for the downstream invocation.

Reply:
Yeah -- sometimes "the argument is not provided" is not the same as "the boolean `false` is provided" :thinking: